### PR TITLE
[SDL-0196] Support Static Icon in Screen Manager

### LINKED ISSUE #1062: [SDL 0196] Add Support for Static Icons to SDLArtwork
Proposal: [Add Support for Static Icons to SDLArtwork](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0196-sdlartwork-static-icons.md)

>Make additions to `SDLArtwork` and `SDLScreenManager` to make using static icons as easy as using uploaded artworks.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/590

Steering Committee Decision:

> The Steering Committee fully agreed to accept this proposal.

### PR DESCRIPTION
**Pull request by joeljfischer**
Fixes #1062 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit tests and smoke tests

### Summary
`SDLArtwork` and screen manager updated to support static icons and the `SDLStaticIconName` enum.

##### Enhancements
* `SDLArtwork` can now be created with a static icon. Such artworks can be passed to the `SDLScreenManager` for use in various image slots such as soft buttons and choices.

##### Bug Fixes
* Fixed failed `SetDisplayLayout` causing `SDLScreenManager` to break.

### Tasks Remaining:
- [x] Unit Tests

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by NicoleYarroch** [comment]
There are two failing test cases in _SDLPreloadChoicesOperationSpec_

File: SmartDeviceLink/SDLSoftButtonManager.m [hunk 1]
```diff
@@ -293,7 +293,7 @@ - (void)sdl_uploadOtherStateImages {
     for (SDLSoftButtonObject *object in self.softButtonObjects) {
         for (SDLSoftButtonState *state in object.states) {
             if ([state.name isEqualToString:object.currentState.name]) { continue; }
-            if (state.artwork != nil && ![self.fileManager hasUploadedFile:state.artwork]) {
+            if (![self sdl_artworkNeedsUpload:object.currentState.artwork]) {
                 [otherStatesToBeUploaded addObject:state.artwork];
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLSoftButtonManager.m]:
```suggestion
              [otherStatesToBeUploaded addObject:object.currentState.artwork];
```

Comment (NicoleYarroch) [reply SmartDeviceLink/SDLSoftButtonManager.m]: Otherwise the app crashes when inserting a `nil` `state.artwork` into the array.

**Comment by NicoleYarroch** [comment]
The `SDLMenuManager` isn't set up to handle sending static artworks. Currently no artworks are sent if one of the menu items has a static icon.

**Comment by NicoleYarroch** [comment]
Maybe you should add static icons to the PICS in the example apps so we have an example of how to use static icons?

File: SmartDeviceLink/SDLSoftButtonManager.m (hunk 1 above)
Comment (joeljfischer) [reply SmartDeviceLink/SDLSoftButtonManager.m]: I think that's incorrect logic that will cause other state artwork to never get uploaded.

Comment (NicoleYarroch) [reply SmartDeviceLink/SDLSoftButtonManager.m]: All I know is that my app kept crashing at that point when I uploaded a soft button with a static icon image

File: Example Apps/Example ObjC/ButtonManager.m [hunk 2]
```diff
@@ -73,7 +73,7 @@ - (void)setToggleEnabled:(BOOL)toggleEnabled {
 }
 
 - (SDLSoftButtonObject *)sdlex_softButtonAlertWithManager:(SDLManager *)manager {
-    SDLSoftButtonState *alertImageAndTextState = [[SDLSoftButtonState alloc] initWithStateName:AlertSoftButtonImageState text:AlertSoftButtonText image:[[UIImage imageNamed:AlertBWIconName] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]];
+    SDLSoftButtonState *alertImageAndTextState = [[SDLSoftButtonState alloc] initWithStateName:AlertSoftButtonImageState text:AlertSoftButtonText artwork:[SDLArtwork artworkWithImage:[[UIImage imageNamed:CarBWIconImageName] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] asImageFormat:SDLArtworkImageFormatPNG]];
```
Comment (NicoleYarroch) [inline Example Apps/Example ObjC/ButtonManager.m]: The Swift example app was not updated with the same example.

**Pull request merged**